src/queries/SearchResultQuery.ts

```diff
@@ -13,26 +13,30 @@ interface SearchResultQueryInterface extends BaseQueryInterface<SearchResult> {
   userId: number;
   keyword?: string;
   url?: string;
+  filterAdUrlWord?: string;
 }
 
 class SearchResultQuery extends BaseQuery<SearchResult> implements SearchResultQueryInterface {
   userId: number;
   keyword?: string;
   url?: string;
+  filterAdUrlWord?: string;
 
   constructor(userId: number, queryParams: SearchResultQueryParams) {
     super({ ...queryParams, databaseModel: dbClient.searchResult });
 
     this.userId = userId;
     this.keyword = queryParams.filters?.keyword;
     this.url = queryParams.filters?.url;
+    this.filterAdUrlWord = queryParams.filters?.filterAdUrlWord;
   }
 
   async call() {
     this.buildUserQueryParams();
     this.buildKeywordQueryParams();
     this.buildUrlToFilterQueryParams();
     this.buildPaginationQueryParams();
+    await this.buildUrlWithWordFilterQueryParams();
```

Comment:
We can rearrange the build functions so that it build the `this.filterQuery` first, `this.paginationQuery` later

```suggestion
    this.buildUserQueryParams();
    this.buildKeywordQueryParams();
    this.buildUrlToFilterQueryParams();
    await this.buildUrlWithWordFilterQueryParams();
    this.buildPaginationQueryParams();
```